src/crystal/system/unix/file_descriptor.cr

```diff
@@ -131,7 +126,7 @@ module Crystal::System::FileDescriptor
 
     # Clear the @volatile_fd before actually closing it in order to
     # reduce the chance of reading an outdated fd value
-    _fd = @volatile_fd.swap(-1)
+    return unless _fd = close_volatile_fd?
```

Comment:
Variables marked with leading underscore are conventionally treated as unused, also there's no need to do that here, why not just drop it?

ditto elsewhere

Reply:
No idea why it was used in the first place, maybe to convey the difference with the `fd` method.

Reply:
It's pretty common to shadow the instance variables / getters, though that's a minor suggestion overall.

Reply:
We probably should just use `fd` because we just poisoned `@volatile_fd` so `#fd` is invalid now.